cardano-cli/src/Cardano/CLI/Shelley/Run.hs

```diff
@@ -12,42 +12,215 @@ module Cardano.CLI.Shelley.Run
+import qualified Shelley.Spec.Ledger.Keys as Ledger
+import qualified Cardano.Crypto.Hash.Class as Crypto
 
 import           Cardano.CLI.Key (VerificationKeyFile(..))
 import           Cardano.CLI.Ops (CliError (..))
-import           Cardano.CLI.Shelley.Parsers (OutputFile (..), ShelleyCommand(..))
+import           Cardano.CLI.Shelley.Parsers
+
+import           Cardano.Config.Shelley.ColdKeys
 import           Cardano.Config.Shelley.KES
-                   (genKESKeyPair, writeKESSigningKey, writeKESVerKey)
+import           Cardano.Config.Shelley.OCert
 import           Cardano.Config.Shelley.VRF
-                   (genVRFKeyPair, writeVRFSigningKey, writeVRFVerKey)
 import           Cardano.Config.Types (SigningKeyFile(..))
 
+
+--
+-- CLI shelley command dispatch
+--
+
 runShelleyClientCommand :: ShelleyCommand -> ExceptT CliError IO ()
-runShelleyClientCommand cc =
-  case cc of
-    ShelleyKeyGenerate fpath -> runShelleyKeyGenerate fpath
-    ShelleyKESKeyPairGenerate vKeyPath sKeyPath duration -> runShelleyKESKeyPairGeneration vKeyPath sKeyPath duration
-    ShelleyVRFKeyPairGenerate vKeyPath sKeyPath -> runShelleyVRFKeyPairGeneration vKeyPath sKeyPath
-    _ -> liftIO . putStrLn $ "runShelleyClientCommand: " ++ show cc
-
-
-runShelleyKeyGenerate :: OutputFile -> ExceptT CliError IO ()
-runShelleyKeyGenerate (OutputFile fpath) = do
-  kp <- liftIO $ shelleyGenKeyPair RegularShelleyKey
-  firstExceptT CardanoApiError $ newExceptT (writeKeyPair fpath kp)
-
-runShelleyKESKeyPairGeneration :: VerificationKeyFile -> SigningKeyFile -> Natural -> ExceptT CliError IO ()
-runShelleyKESKeyPairGeneration (VerificationKeyFile vKeyPath) (SigningKeyFile sKeyPath) duration = do
-  (vKESKey, sKESKey) <- liftIO $ genKESKeyPair duration
-  firstExceptT KESCliError $ writeKESSigningKey sKeyPath sKESKey
-  firstExceptT KESCliError $ writeKESVerKey vKeyPath vKESKey
-
-runShelleyVRFKeyPairGeneration :: VerificationKeyFile -> SigningKeyFile  -> ExceptT CliError IO ()
-runShelleyVRFKeyPairGeneration (VerificationKeyFile vKeyPath) (SigningKeyFile sKeyPath) = do
-  (sKESKey, vKESKey) <- liftIO $ genVRFKeyPair
-  firstExceptT VRFCliError $ writeVRFSigningKey sKeyPath sKESKey
-  firstExceptT VRFCliError $ writeVRFVerKey vKeyPath vKESKey
+runShelleyClientCommand (AddressCmd      cmd) = runAddressCmd      cmd
+runShelleyClientCommand (StakeAddressCmd cmd) = runStakeAddressCmd cmd
+runShelleyClientCommand (TransactionCmd  cmd) = runTransactionCmd  cmd
+runShelleyClientCommand (NodeCmd         cmd) = runNodeCmd         cmd
+runShelleyClientCommand (PoolCmd         cmd) = runPoolCmd         cmd
+runShelleyClientCommand (QueryCmd        cmd) = runQueryCmd        cmd
+runShelleyClientCommand (BlockCmd        cmd) = runBlockCmd        cmd
+runShelleyClientCommand (SystemCmd       cmd) = runSystemCmd       cmd
+runShelleyClientCommand (DevOpsCmd       cmd) = runDevOpsCmd       cmd
+runShelleyClientCommand (GenesisCmd      cmd) = runGenesisCmd      cmd
```

Comment:
I think alignment like this is a mistake. With alignment like this adding a new constructor or removing the one with the longest name, will result in them all being reformatted. The resulting diff will then contain semantically null whitespace changes,

If we avoid the indentation, then follow up diffs will be minimal.

Reply:
You can ignore whitespace changes in diffs, even GitHub supports that.

Reply:
I realise this is a matter of opinion and reasonable people can and do differ.

I think on balance I've concluded that the readability benefits of nicely formatted code generally outweigh the diff noise.

Reply:
Sure, but I needed to add an extra parameter to one of those commands and adding that parameter screwed up the formatting.